pkg/appliance/backup.go

```diff
@@ -216,14 +216,19 @@ func PerformBackup(cmd *cobra.Command, args []string, opts *BackupOpts) (map[str
 
 	retryStatus := func(ctx context.Context, applianceID, backupID string) error {
 		return backoff.Retry(func() error {
-			status, err := backupAPI.Status(ctx, applianceID, backupID)
-			if err != nil {
-				return err
-			}
-			if status != backup.Done {
-				return fmt.Errorf("Backup not done for appliance %s, got %s", applianceID, status)
+			select {
+			case <-ctx.Done():
+				return backoff.Permanent(ctx.Err())
+			default:
+				status, err := backupAPI.Status(ctx, applianceID, backupID)
+				if err != nil {
+					return err
+				}
+				if status != backup.Done {
+					return fmt.Errorf("Backup not done for appliance %s, got %s", applianceID, status)
+				}
+				return nil
 			}
-			return nil
 		}, backoff.NewExponentialBackOff())
```

Comment:
I think we could simplify this by using backoff.WithContext
```diff
diff --git a/pkg/appliance/backup.go b/pkg/appliance/backup.go
index 52e3f7d..b35a944 100644
--- a/pkg/appliance/backup.go
+++ b/pkg/appliance/backup.go
@@ -224,7 +224,7 @@ func PerformBackup(cmd *cobra.Command, args []string, opts *BackupOpts) (map[str
 				return fmt.Errorf("Backup not done for appliance %s, got %s", applianceID, status)
 			}
 			return nil
-		}, backoff.NewExponentialBackOff())
+		}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
 	}
 
 	b := func(appliance openapi.Appliance) (backedUp, error) {

```

Reply:
That makes a whole lot more sense.